File: backend/src/main/java/es/udc/paproject/backend/model/services/MilkingService.java
package es.udc.paproject.backend.model.services;

import es.udc.paproject.backend.model.entities.Milking;
import es.udc.paproject.backend.model.exceptions.InstanceNotFoundException;
import es.udc.paproject.backend.model.exceptions.PermissionException;
import es.udc.paproject.backend.rest.dtos.MilkingDTOs.MilkingChartDto;
import es.udc.paproject.backend.rest.dtos.MilkingDTOs.MilkingDto;

import java.util.List;

public interface MilkingService {

    void createMilking(Milking milking) throws InstanceNotFoundException;

    void updateMilking(Long id, Integer liters) throws InstanceNotFoundException;

    void deleteMilkingRecord(Long userId, Long id) throws InstanceNotFoundException, PermissionException;

    Milking getMilkingById(Long id) throws InstanceNotFoundException;

    Block<Milking> getAllMilkings(Long userId, Long animalId, Integer startLiters, Integer endLiters, String startDate, String endDate,
                                  int page, int size) throws InstanceNotFoundException;

    List<MilkingChartDto> findMilkingByAnimalMilkedId(Long userId, Long animalId) throws InstanceNotFoundException;
}

File: frontend/src/backend/milkingService.js
import {config, appFetch} from "./appFetch";

export const registerMilking = (milking, onSuccess, onErrors) => {
    appFetch('/milking/registerMilking', config('POST', milking), onSuccess, onErrors);
}

export const updateMilking = (milkingId, milking, onSuccess, onErrors) => {
    appFetch(`/milking/${milkingId}`, config('PUT', milking), onSuccess, onErrors);
}

export const deleteMilking = (milkingId, onSuccess, onErrors) => {
    appFetch(`/milking/${milkingId}`, config('DELETE'), onSuccess, onErrors);
}

export const getMilkingById = (milkingId, onSuccess, onErrors) => {
    appFetch(`/milking/${milkingId}`, config('GET'), onSuccess, onErrors);
}

export const getAllMilkings = ({page, animalId, startLiters, endLiters, startDate, endDate}, onSuccess, onErrrors) => {
    let path = `/milking/AllMilkingByFarmId?page=${page}&size=5`;

    path += animalId != null && animalId > 0 ? `&animalId=${encodeURIComponent(animalId)}` : "";
    path += startLiters != null && startLiters > 0 ? `&startLiters=${encodeURIComponent(startLiters)}`: "";
    path += endLiters != null && endLiters > 0 ? `&endLiters=${encodeURIComponent(endLiters)}` : "";
    path += startDate ? `&startDate=${encodeURIComponent(startDate.toISOString().substring(0,10))}` : "";
    path += endDate ? `&endDate=${encodeURIComponent(endDate.toISOString().substring(0,10))}` : "";

    appFetch(path, config('GET'), onSuccess, onErrrors);
}